### DISCUSSION TITLE
MATCH_SSH_KEY not correct need help

### DISCUSSION THREAD

**Discussion by Nicholas-Rotondo** [codemagic-ci-cd/codemagic-docs#2666] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

iOS (Swift, Objective-C)

### Steps to reproduce

1. Currently took over an ios app that was being deployed using codemagic. Past dev gave me a bunch of key properties. In these key properties everything was working but the only one I know isn't working is MATCH_SSH_KEY. This dev has departed and after going through all the files and using Claude to help I realized in the Matchfile he was using a personal github that held some certificates(can't see them as they are either deleter or private). Now im using our company's bitbucket url in the matchfile which the fastfile checks. However, when the past dev was here all of my deployments were good but now they have the error of 403 saying i dont have permission. When i tried to use a new MATCH_SSH_KEY the build either completed way too fast(deploy was under 5s) or i received an error saying 403 or return 128 instead of 0. Either way im very lost. I can promise you I have tried every possible solution i could find and used claude of chat to help me solve this. Now a couple questions i have
2. What format is the MATCH_SSH_KEY supposed to be in? In his config file I noticed it is in base64 but when i check the site it doesn't mention anything along those lines. When i generate a private key it doesn't work for some reason either. 

### Expected results

Deployment is supposed to be correct but it throws errors mentioned above. 

### Actual results

Errors 403 or return 128 instead of 0.

**Comment by dam-ease** [comment]
Hello @Nicholas-Rotondo,
The `MATCH_SSH_KEY` should begin with:
`-----BEGIN OPENSSH PRIVATE KEY-----`
and end with `-----END OPENSSH PRIVATE KEY-----`
You can also check out [this documentation](https://docs.codemagic.io/knowledge-git/access-private-git-submodules/)
If these steps have been followed and you still run into issues, please share the build ID with me? You can find it under your build details page's 'ID' section. ﻿Navigate to the 'Builds' section on the left-hand menu, select the build you want to view, and click the copy icon.

**Reply by Nicholas-Rotondo** [reply]
Hey sorry for the late reply, here is the build id: 66b6122c3f6a6f0ff9a780ba @dam-ease
